ocp_resources/cron_job.py

```diff
@@ -0,0 +1,92 @@
+
+
+class CronJob(NamespacedResource):
+    """
+    CronJob object. API reference:
+    https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/
+    """
+
+    api_group = NamespacedResource.ApiGroup.BATCH
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        client=None,
+        teardown=True,
+        schedule=None,
+        job_template=None,
+        timezone=None,
+        concurrency_policy=None,
+        starting_deadline_seconds=None,
+        suspend=None,
+        successful_jobs_history_limit=None,
+        failed_jobs_history_limit=None,
+        privileged_client=None,
+        yaml_file=None,
+        delete_timeout=TIMEOUT_4MINUTES,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the endpoints resource
+            namespace (str): Namespace of endpoints resource
+            client: (DynamicClient): DynamicClient for api calls
+            teardown (bool): Indicates if the resource should be torn down at the end
+            privileged_client (DynamicClient): Privileged client for api calls
+            yaml_file (str): yaml file for the resource.
+            delete_timeout (int): timeout associated with delete action
+            schedule (str): schedule of the cron job
+            job_template (JobTemplateSpec): describes the job that would be created when a cronjob would be executed
+            timezone (str): timezone name for the given schedule
+            concurrency_policy (str): indicates how to treat concurrent execution of a job
+            suspend (bool): suspend subsequent executions
+            successful_jobs_history_limit (int): number of successful jobs to retain
+            failed_jobs_history_limit (int): number of failed jobs to retain
+            starting_deadline_seconds (int): deadline in seconds, for starting a job, in case it does not start at
+                                             scheduled time
+
+        Example:
+            job_template: https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/job-v1/#JobSpec
+        """
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            teardown=teardown,
+            privileged_client=privileged_client,
+            yaml_file=yaml_file,
+            delete_timeout=delete_timeout,
+            **kwargs,
+        )
+        self.job_template = job_template
+        self.schedule = schedule
+        self.timezone = timezone
+        self.concurrency_policy = concurrency_policy
+        self.suspend = suspend
+        self.successful_jobs_history_limit = successful_jobs_history_limit
+        self.failed_jobs_history_limit = failed_jobs_history_limit
+        self.starting_deadline_seconds = starting_deadline_seconds
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            if not (self.job_template and self.schedule):
```

Comment:
what about the other ones? (concurrency_policy,failed_jobs_history_limit, etc)
either add to validation or mark as optional in the docstring

Reply:
None of those are required. Only job_template and schedule are required. But I would need to update docstring to reflect that.